collect_app/src/main/java/org/odk/collect/android/widgets/BaseStringWidget.java

```diff
@@ -108,4 +109,25 @@ public void afterTextChanged(Editable s) {
             }
         });
     }
+
+    protected void setUpIntegerInputFilter(boolean useThousandSeparator) {
```

Comment:
It doesn't feel great to have these methods at the top of the hierarchy when they only are relevant to certain subclasses.

In general, the hierarchy here has made me feel a bit uneasy because it's strange to have numeric types extend a text type. Perhaps you'd consider calling the new base type something like `EditTextWidget` or `UserInputWidget` or something like that? I haven't thought deeply about the name but hopefully you see what I'm getting at -- what groups these widgets together is that they all have a text box for the user to fill out.